test/BitcoinTx.t.sol

```diff
@@ -43,12 +43,125 @@ contract BitcoinTxTest is Test {
+    //    function test_ensureTxInputSpendsUtxoEmptyBytes() public {
+    //        bytes memory inputVector = new bytes(0);
+    //        BitcoinTx.UTXO memory utxo;
+    //        BitcoinTx.ensureTxInputSpendsUtxo(inputVector,utxo);
+    //    }
+
+    //
+    // end ensureTxInputSpendsUtxo
+    //
+
+    //
+    // start getTxOutputValue
+    //
+    function test_getTxOutputValue() public {
         // b1273a6c00eba20ee8837e445599d1362e005f6e1a8525802ba57bc515461a3a
         uint64 value = BitcoinTx.getTxOutputValue(
             keccak256(hex"16001493adab0a7a8cb7675db135c9c97e81942025c2c9"),
             hex"02c67d16000000000016001493adab0a7a8cb7675db135c9c97e81942025c2c9aea79b4200000000160014f60834ef165253c571b11ce9fa74e46692fc5ec1"
         );
         assertEq(value, 1473990);
     }
+
+    function test_getTxOutputValueSingleOutputTransaction() public {
+        // tx api: https://btc-testnet.gobob.xyz/tx/b0fe4bd36b17be89f131c2e652578def3cc0c3d5aa9e7a3f972365a8dc46dba8
+        uint64 value = BitcoinTx.getTxOutputValue(
+            keccak256(hex"160014d127b24a7e2aad2ddf21d4d940f6202158aa507d"),
+            hex"01ab9ab90800000000160014d127b24a7e2aad2ddf21d4d940f6202158aa507d"
+        );
+        assertEq(value, 146381483);
+    }
+
+    function test_getTxOutputValueMaxTransactionValue() public {
+        // tx api: https://api.blockcypher.com/v1/btc/main/txs/d486aeb0e59181fd1addb4aa69ce04d638188fc1125c424899267e8ed6a8af24?limit=50&includeHex=true/
+        uint64 value = BitcoinTx.getTxOutputValue(
+            keccak256(hex"17a914ed498d84acb4532656fcf6947d0ceab6c77188bc87"),
+            hex"0260536280ed03000017a9148e097444bb754122652208bf00f71a87b177b700874b5a115e2704000017a914ed498d84acb4532656fcf6947d0ceab6c77188bc87"
+        );
+        assertEq(value, 4567128431179);
+    }
+
+    function test_getTxOutputValueWithInvalidPubkey() public {
+        vm.expectRevert("No output found for scriptPubKey");
+        BitcoinTx.getTxOutputValue(
+            keccak256(hex"000000000000000000000000000000000000000000000000"),
+            hex"02c67d16000000000016001493adab0a7a8cb7675db135c9c97e81942025c2c9aea79b4200000000160014f60834ef165253c571b11ce9fa74e46692fc5ec1"
+        );
+    }
+
+    // Panic: [FAIL. Reason: panic: array out-of-bounds access (0x32)]
+    //    function test_getTxOutputValueEmptyBytes() public {
+    //        bytes memory emptyBytes = new bytes(0);
+    //        uint64 value = BitcoinTx.getTxOutputValue(
+    //            keccak256(hex"16001493adab0a7a8cb7675db135c9c97e81942025c2c9"),
+    //            emptyBytes
+    //        );
+    //    }
+
+    // [FAIL. Reason: EvmError: OutOfGas]
+    //    function test_getTxOutputValueWithInvalidInput() public {
+    //        uint64 value = BitcoinTx.getTxOutputValue(
+    //            keccak256(hex"16001493adab0a7a8cb7675db135c9c97e81942025c2c9"),
+    //            hex"16001493adab0a7a8cb7675db135c9c97e81942025c2c9"
+    //        );
+    //    }
```

Comment:
during debugging found out that, this happens because the `outputLength`  calculations emit a huge value since the wrong output is given. 
Therefore the assembly calculation [here](https://github.com/bob-collective/bob/blob/master/src/bridge/BitcoinTx.sol#L272-L280) exhausts the gas. 
Not sure want to handle this, as estimated gas will show a huge value if the inputs are wrong.